lightning-invoice/src/de.rs

```diff
@@ -585,12 +587,16 @@ impl FromBase32 for Route {
 			let mut channel_id: [u8; 8] = Default::default();
 			channel_id.copy_from_slice(&hop_bytes[33..41]);
 
-			let hop = RouteHop {
-				pubkey: PublicKey::from_slice(&hop_bytes[0..33])?,
-				short_channel_id: channel_id,
-				fee_base_msat: parse_int_be(&hop_bytes[41..45], 256).expect("slice too big?"),
-				fee_proportional_millionths: parse_int_be(&hop_bytes[45..49], 256).expect("slice too big?"),
-				cltv_expiry_delta: parse_int_be(&hop_bytes[49..51], 256).expect("slice too big?")
+			let hop = RouteHint {
+				src_node_id: PublicKey::from_slice(&hop_bytes[0..33])?,
+				short_channel_id: u64::from_be_bytes(channel_id),
```

Comment:
I think you can just swap this for `parse_int_be` like the other integer fields below.